File: testbed_php/test/attributes/ConstraintTest.php
<?php
class ConstraintTest extends UnitTestCase
{

  public function test_checkConstraint()
  {
    $boat = new BoatA(20);
    $this->assertEqual(false, $boat->setAge(18));
    $this->assertEqual(true, $boat->setAge(19));
  }

  public function test_checkNegation()
  {
    $boat = new BoatB(2);
    $this->assertEqual(true, $boat->setAge(18));
    $this->assertEqual(false, $boat->setAge(19));
  }

}

File: testbed_python/test/attributes/constraint_test.py
import unittest

# Test classes are imported dynamically
# Must be generated into ImportModules' namespace then imported
from ImportModules import importModules

importModules(["BoatA", "BoatB"], ["attributes", "test"])
from ImportModules import *


class ConstraintTest(unittest.TestCase):
    def test_checkConstraint(self):
        boat = BoatA.BoatA(20)
        self.assertEqual(False, boat.setAge(18))
        self.assertEqual(True, boat.setAge(19))

    def test_checkNegation(self):
        boat = BoatB.BoatB(2)
        self.assertEqual(True, boat.setAge(18))
        self.assertEqual(False, boat.setAge(19))
